src/stdatomic.h

```diff
@@ -0,0 +1,163 @@
+#ifndef __MSVC_STDATOMIC_H
```

Comment:
This atomic implementation is for msvc to build with c compiler, since windows sdk does not include a C atomic implementation (mingw should be configured to exclude this).

Code from github repo:
https://github.com/zenny-chen/simple-stdatomic-for-VS-Clang
under apache 2.0 license.
(maybe add a license note?)